code/obj/item/uplinks.dm

```diff
@@ -1334,23 +1334,24 @@ Note: Add new traitor items to syndicate_buylist.dm, not here.
 		.["vr"] = src.vr
 
 		var/list/spellbook_contents = list()
-		for(var/datum/SWFuplinkspell/spell in src.spells)
+		for(var/datum/SWFuplinkspell/spell as anything in src.spells)
 			var/cooldown_contents = null
 			var/icon/spell_icon = null
-			if (spell.eqtype != "Spell") // Disallow spell framework
-				if (!spellbook_contents[spell.eqtype]) spellbook_contents[spell.eqtype] = list() // Create category if it doesnt exist
-				if (spell.assoc_spell && ispath(spell.assoc_spell, /datum/targetable/spell))
-					var/datum/targetable/spell/spell_datum = spell.assoc_spell
-					cooldown_contents = initial(spell_datum.cooldown)
-					spell_icon = icon(initial(spell_datum.icon), initial(spell_datum.icon_state), frame=6)
-				spellbook_contents[spell.eqtype][spell.name] = list(
-					desc = spell.desc,
-					cost = spell.cost,
-					cooldown = cooldown_contents,
-					vr_allowed = spell.vr_allowed,
-					spell_img = icon2base64(spell_icon)
-				)
-
+			if (!spellbook_contents[spell.eqtype])
+				// create category if it doesnt exist
+				spellbook_contents[spell.eqtype] = list()
+			if (spell.assoc_spell && ispath(spell.assoc_spell, /datum/targetable/spell))
+				var/datum/targetable/spell/spell_ability_datum = spell.assoc_spell
+				cooldown_contents = initial(spell_ability_datum.cooldown)
+				spell_icon = icon2base64(icon(initial(spell_ability_datum.icon), initial(spell_ability_datum.icon_state), frame=6))
+			spellbook_contents[spell.eqtype] += list(list(
+				cooldown = cooldown_contents,
```

Comment:
Consider changing the units of this to seconds, to allow for display as provided.  Would also remove seemingly arbitrary conversion from deciseconds.

Reply:
Port of existing, but valid.